File: admin/src/api/erp_payment_detail.ts
import { PaginatedRequest, PaginatedResponse } from "@/base/page";
import { api } from "@/utils/request";

const apis = {
  create: "/erp/erp_payment_detail/create", // 新增
  update: "/erp/erp_payment_detail/update", // 修改
  delete: "/erp/erp_payment_detail/delete", // 删除
  get: "/erp/erp_payment_detail/get", // 单条查询
  list: "/erp/erp_payment_detail/list", // 列表查询
  page: "/erp/erp_payment_detail/page", // 分页查询
};

export interface ErpPaymentDetailRequest {
  id?: number; // 付款详情ID
  purchase_order_id?: number; // 采购订单ID
  purchase_return_id?: number; // 采购退货ID
  amount: number; // 金额
  remarks: string; // 备注

  type?: number; // 0为采购订单入库,1为采购退货
}

export interface ErpPaymentDetailResponse {
  id: number; // 付款详情ID
  payment_id: number; // 付款ID
  purchase_order_id: number; // 采购订单ID
  purchase_return_id: number; // 采购退货ID
  amount: number; // 金额
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
}

export interface ErpPaymentDetailBaseResponse {
  id: number; // 付款详情ID
  purchase_order_id: number; // 采购订单ID
  purchase_return_id: number; // 采购退货ID
  amount: number; // 金额
  remarks: string; // 备注

  type?: number; // 0为采购订单入库,1为采购退货
}

export interface ErpPaymentDetailQueryCondition extends PaginatedRequest {}

export const createErpPaymentDetail = (
  erp_payment_detail: ErpPaymentDetailRequest
): Promise<number> => {
  return api.post<number>(apis.create, erp_payment_detail);
};

export const updateErpPaymentDetail = (
  erp_payment_detail: ErpPaymentDetailRequest
): Promise<void> => {
  return api.post<void>(apis.update, erp_payment_detail);
};

export const deleteErpPaymentDetail = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
};

export const getErpPaymentDetail = (
  id: number
): Promise<ErpPaymentDetailResponse> => {
  return api.get<ErpPaymentDetailResponse>(`${apis.get}/${id}`);
};

export const listErpPaymentDetail = (): Promise<
  Array<ErpPaymentDetailResponse>
> => {
  return api.get<Array<ErpPaymentDetailResponse>>(apis.list);
};

export const pageErpPaymentDetail = (
  condition: ErpPaymentDetailQueryCondition
): Promise<PaginatedResponse<ErpPaymentDetailResponse>> => {
  return api.get<PaginatedResponse<ErpPaymentDetailResponse>>(
    apis.page,
    condition
  );
};

File: server/module/module-erp/erp-server/src/convert/erp_payment_detail.rs
use sea_orm::{Set, NotSet};
use crate::model::erp_payment_detail::{self, Model as ErpPaymentDetail, ActiveModel as ErpPaymentDetailActiveModel};
use erp_model::request::erp_payment_detail::{CreateErpPaymentDetailRequest, UpdateErpPaymentDetailRequest};
use erp_model::response::erp_payment_detail::{ErpPaymentDetailBaseResponse, ErpPaymentDetailResponse};

pub fn create_request_to_model(request: &CreateErpPaymentDetailRequest) -> ErpPaymentDetailActiveModel {
    ErpPaymentDetailActiveModel {
        purchase_order_id: request.purchase_order_id.as_ref().map_or(NotSet, |purchase_order_id| Set(Some(purchase_order_id.clone()))),
        purchase_return_id: request.purchase_return_id.as_ref().map_or(NotSet, |purchase_return_id| Set(Some(purchase_return_id.clone()))),
        amount: Set(request.amount.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_add_request_to_model(request: &UpdateErpPaymentDetailRequest) -> ErpPaymentDetailActiveModel {
    ErpPaymentDetailActiveModel {
        purchase_order_id: request.purchase_order_id.as_ref().map_or(NotSet, |purchase_order_id| Set(Some(purchase_order_id.clone()))),
        purchase_return_id: request.purchase_return_id.as_ref().map_or(NotSet, |purchase_return_id| Set(Some(purchase_return_id.clone()))),
        amount: Set(request.amount.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateErpPaymentDetailRequest, existing: ErpPaymentDetail) -> ErpPaymentDetailActiveModel {
    let mut active_model: ErpPaymentDetailActiveModel = existing.into();
    if let Some(purchase_order_id) = &request.purchase_order_id { 
        active_model.purchase_order_id = Set(Some(purchase_order_id.clone()));
    }
    if let Some(purchase_return_id) = &request.purchase_return_id { 
        active_model.purchase_return_id = Set(Some(purchase_return_id.clone()));
    }
    if let amount = &request.amount { 
        active_model.amount = Set(amount.clone());
    }
    if let Some(remarks) = &request.remarks { 
        active_model.remarks = Set(Some(remarks.clone()));
    }
    active_model
}

pub fn model_to_response(model: ErpPaymentDetail) -> ErpPaymentDetailResponse {
    ErpPaymentDetailResponse { 
        id: model.id,
        payment_id: model.payment_id,
        purchase_order_id: model.purchase_order_id,
        purchase_return_id: model.purchase_return_id,
        amount: model.amount,
        remarks: model.remarks,
        department_code: model.department_code,
        department_id: model.department_id,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

pub fn model_to_base_response(model: ErpPaymentDetail) -> ErpPaymentDetailBaseResponse {
    ErpPaymentDetailBaseResponse { 
        id: model.id,
        purchase_order_id: model.purchase_order_id,
        purchase_return_id: model.purchase_return_id,
        amount: model.amount,
        remarks: model.remarks,
    }
}